src/Build.UnitTests/BackEnd/TaskExecutionHost_Tests.cs

```diff
@@ -1029,6 +1029,24 @@ public void TestTaskResolutionFailureWithNoUsingTask()
             _logger.AssertLogContains("MSB4036");
         }
 
+        /// <summary>
+        /// https://github.com/dotnet/msbuild/issues/8864
+        /// </summary>
+        [Fact]
+        public void TestTaskDictionaryOutputItems()
```

Comment:
Can this somehow test as well that the output has the expected metadata (to verify that the fix aids to properly copying of the metadata)